templates/backup.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+# Logic is as follows:
+
+# 1. If pre-script fails exit immediately
+# 2. Proceed if backup
+# 3. Irrespective whether backup succeeds execute post-script.
+
+timestamp=$(date -Isecond)
+
+{% if TARSNAP_BACKUP_PRE_SCRIPT %}
+{{ TARSNAP_BACKUP_PRE_SCRIPT }} || exit 1
+{% endif %}
+
+# We run --fsck before backup, so in case of any inconsistencies (that might happen)
+# backups will still work
+
+/usr/local/bin/tarsnap --fsck --keyfile "{{ TARSNAP_KEY_REMOTE_LOCATION }}" \
```

Comment:
@jbzdak I'd rather not do this before we found a need for it - if these operations were required tarsnap would do it every time it is run.